pkg/sriov/sriov.go

```diff
@@ -133,49 +161,67 @@ func getBondMembers(bond netlink.Link) ([]string, error) {
 	return result, nil
 }
 
-// GetNetRepresentor retrieves network representor device for smartvf
+// GetNetRepresentor returns representor name for passed device ID. Supported devices are Virtual Function
+// or Scalable Function
 func GetNetRepresentor(deviceID string) (string, error) {
-	// get Uplink netdevice.  The uplink is basically the PF name of the deviceID (smart VF).
-	// The uplink is later used to retrieve the representor for the smart VF.
-	uplink, err := sriovnet.GetUplinkRepresentor(deviceID)
-	if err != nil {
-		return "", err
-	}
+	var rep, uplink string
+	var err error
+	var index int
 
-	// get smart VF index from PCI
-	vfIndex, err := sriovnet.GetVfIndexByPciAddress(deviceID)
-	if err != nil {
-		return "", err
+	if IsPCIDeviceName(deviceID) { // PCI device
+		uplink, err = sriovnet.GetUplinkRepresentor(deviceID)
+		if err != nil {
+			return "", err
+		}
+		index, err = sriovnet.GetVfIndexByPciAddress(deviceID)
+		if err != nil {
+			return "", err
+		}
+		rep, err = sriovnet.GetVfRepresentor(uplink, index)
+	} else if IsAuxDeviceName(deviceID) { // Auxiliary device
+		uplink, err = sriovnet.GetUplinkRepresentorFromAux(deviceID)
+		if err != nil {
+			return "", err
+		}
+		index, err = sriovnet.GetSfIndexByAuxDev(deviceID)
+		if err != nil {
+			return "", err
+		}
+		rep, err = sriovnet.GetSfRepresentor(uplink, index)
+	} else {
+		return "", fmt.Errorf("cannot determine device type for id '%s'", deviceID)
 	}
-
-	// get smart VF representor interface. This is a host net device which represents
-	// smart VF attached inside the container by device plugin. It can be considered
-	// as one end of veth pair whereas other end is smartVF. The VF representor would
-	// get added into ovs bridge for the control plane configuration.
-	rep, err := sriovnet.GetVfRepresentor(uplink, vfIndex)
 	if err != nil {
 		return "", err
 	}
-
 	return rep, nil
 }
 
 // SetupSriovInterface moves smartVF into container namespace, rename it with ifName and also returns host interface with VF's representor device
 func SetupSriovInterface(contNetns ns.NetNS, containerID, ifName string, mtu int, deviceID string) (*current.Interface, *current.Interface, error) {
```

Comment:
should we rename this method as it now handles both sr-iov and auxiliary devices ?